optimum/habana/transformers/models/mpt/modeling_mpt.py

```diff
@@ -90,13 +126,38 @@ def gaudi_mpt_attention_forward(
     if attention_mask is not None:
         attention_scores = attention_scores.masked_fill(attention_mask, torch.finfo(query_states.dtype).min)
 
-    # (batch_size, n_heads, seq_length, key_length)
-    attn_weights = nn.functional.softmax(attention_scores.float(), dim=-1).to(value_states.dtype)
-    attn_weights = nn.functional.dropout(attn_weights, p=self.attn_dropout_p, training=self.training)
+    if use_flash_attention and FusedSDPA:
+        import habana_frameworks.torch.hpu as ht
+
+        with ht.sdp_kernel(enable_recompute=flash_attention_recompute):
+            if query_length > 8192:
+                attn_output = self.gaudi_flash_attn_v1(
```

Comment:
where is self.gaudi_flash_attn_v1 defined?

Reply:
I followed qwen2, and didn't find in any other models. Looks it is not needed in this model, so I removed it